src/Builder/Encoder/PipelineEncoder.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Builder\Encoder;
+
+use MongoDB\Builder\Pipeline;
+use MongoDB\Codec\EncodeIfSupported;
+use MongoDB\Exception\UnsupportedValueException;
+use stdClass;
+
+/** @template-extends AbstractExpressionEncoder<array, Pipeline> */
+class PipelineEncoder extends AbstractExpressionEncoder
+{
+    /** @template-use EncodeIfSupported<array, Pipeline> */
+    use EncodeIfSupported;
+
+    public function canEncode(mixed $value): bool
+    {
+        return $value instanceof Pipeline;
+    }
+
+    public function encode(mixed $value): stdClass|array|string
+    {
+        if (! $this->canEncode($value)) {
```

Comment:
We need a method `encodeIfSupported` to avoid repeating this 3 lines in all encoders.